packages/client/lib/functions/placeShip.tsx

```diff
@@ -29,67 +31,53 @@ const placeShip = (
       console.log("goesOutsideGrid");
       return true;
     }
-    return false;
+    return undefined;
   };
 
   /** True if neighbouring x or y + ... ship length coordinates is not occupied + !isOutsideGrid */
   const isLocationPlaceable = (x: number, y: number) => {
-    if (goesOutsideGrid()) return false;
-
     if (orientation === "horizontal") {
       for (let i = x; i < xrange; i++) {
-        if (board[i][y] !== 0) {
-          return false;
+        console.log("Checking for ship at " + i + ", " + y);
+        const foundShips = shipsInCoord(shipInfo, [i, y]);
+        if (foundShips) {
+          console.log("shipsFound", foundShips);
+          return undefined;
         }
       }
-    }
-    if (orientation === "vertical") {
+    } else {
       for (let j = y; j < yrange; j++) {
-        if (board[x][j] !== 0) {
-          return false;
+        const foundShips = shipsInCoord(shipInfo, [x, j]);
+        if (foundShips) {
+          console.log("shipsFound", foundShips);
+
+          return undefined;
         }
       }
     }
+
     return true;
   };
 
-  /** Updates both the placedLocation and the board */
-  const updateShipPlacement = (x: number, y: number) => {
-    setPlayerData((prev) => {
```

Comment:
I moved this out of here to `generateRandomPlacement` to call the function once instead of multiple times in here, but I JUST realized that this breaks manual placing. Fix for me ty :D